pkg/controller/windowsmachine/metrics/metrics.go

```diff
@@ -229,3 +234,25 @@ func isEndpointsValid(nodes *v1.NodeList, endpoints *v1.Endpoints) bool {
 	}
 	return true
 }
+
+// updateServiceMonitors patches the metrics Service Monitor to include metrics path and replace action that replaces
+// node ip addressed returned by the windows exporter to node names. This is required to display node graphs on the
```

Comment:
Not sure what you are trying to say here, especially `replace action that replaces`. Also are you saying  `node IP addresses returned by windows_exporter with node names`?

Reply:
I will update the comment.

Console graph queries require metrics endpoint target name to be `node name`, however `windows_exporter` returns `node IP`. We replace the target name by adding following to the `ServiceMonitor`

```
      path: "/metrics"
      relabelings:
      - action: replace
        regex: (.*)
        replacement: $1
        sourceLabels:
        - __meta_kubernetes_endpoint_address_target_name
        targetLabel: instance
```

Reply:
Please add this level of detail to the comment.